ava-labs/core-mobile
# CP-8632 Fix failing swap and excesive showing wallet connect warning dialog

## Description

**Ticket: [CP-8632]** 

Problem seemed to be in wrongly defined function which overrides `window.ethereum` object.
Also we now check if connection with dApp is made and if so we won't show "Connect with WC" warning modal, b/c some dApps call `window.ethereum` object even if connected with WC.


## Screenshots/Videos

https://github.com/ava-labs/avalanche-wallet-apps/assets/6573904/a7683504-8890-4559-bb27-c788012dcb9e


## Testing
Steps:

- open in-app browser and navigate to core.app
- connect your wallet
- select Swap from + menu
- select AVAX as the from token
- Select USDC as the to token
- Enter .01 or similar for the amount to swap from
- Click on Swap button
= should successfully swap

## Checklist

Please check all that apply (if applicable)


[CP-8632]: https://ava-labs.atlassian.net/browse/CP-8632?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
